tests/test_modeling_common.py

```diff
@@ -619,6 +619,41 @@ def test_training_gradient_checkpointing(self):
             loss = model(**inputs).loss
             loss.backward()
 
+    @slow
+    @pytest.mark.gradient_checkpointing_autocast_test
+    def test_training_gradient_checkpointing_autocast(self):
```

Comment:
Small note on the test. I have ran the tests on all transformers : 

```bash
FAILED tests/models/align/test_modeling_align.py::AlignTextModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param pooler.dense.weight
FAILED tests/models/altclip/test_modeling_altclip.py::AltCLIPVisionModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vision_model.post_layernorm.weight
FAILED tests/models/autoformer/test_modeling_autoformer.py::AutoformerModelTest::test_training_gradient_checkpointing_autocast - RuntimeError: cuFFT only supports dimensions whose sizes are powers of two when computing in half precision, but got a signal size of[14]
FAILED tests/models/beit/test_modeling_beit.py::BeitModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param beit.encoder.layer.2.lambda_1
FAILED tests/models/big_bird/test_modeling_big_bird.py::BigBirdModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param bert.pooler.weight
FAILED tests/models/blip/test_modeling_blip.py::BlipTextRetrievalModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vision_proj.weight
FAILED tests/models/canine/test_modeling_canine.py::CanineModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param canine.projection.conv.weight
FAILED tests/models/chinese_clip/test_modeling_chinese_clip.py::ChineseCLIPTextModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param pooler.dense.weight
FAILED tests/models/chinese_clip/test_modeling_chinese_clip.py::ChineseCLIPVisionModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vision_model.post_layernorm.weight
FAILED tests/models/clip/test_modeling_clip.py::CLIPVisionModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vision_model.post_layernorm.weight
FAILED tests/models/clipseg/test_modeling_clipseg.py::CLIPSegVisionModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vision_model.post_layernorm.weight
FAILED tests/models/clipseg/test_modeling_clipseg.py::CLIPSegModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param clip.logit_scale
FAILED tests/models/data2vec/test_modeling_data2vec_vision.py::Data2VecVisionModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param data2vec_vision.encoder.layer.2.lambda_1
FAILED tests/models/dpt/test_modeling_dpt.py::DPTModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param dpt.layernorm.weight
FAILED tests/models/dpt/test_modeling_dpt_hybrid.py::DPTModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param dpt.layernorm.weight
FAILED tests/models/flava/test_modeling_flava.py::FlavaImageModelTest::test_training_gradient_checkpointing_autocast - AttributeError: 'FlavaImageModelTester' object has no attribute 'is_training'
FAILED tests/models/flava/test_modeling_flava.py::FlavaTextModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param pooler.dense.weight
FAILED tests/models/flava/test_modeling_flava.py::FlavaMultimodalModelTest::test_training_gradient_checkpointing_autocast - AttributeError: 'FlavaMultimodalModelTester' object has no attribute 'is_training'
FAILED tests/models/flava/test_modeling_flava.py::FlavaImageCodebookTest::test_training_gradient_checkpointing_autocast - AttributeError: 'FlavaImageCodebookTester' object has no attribute 'is_training'
FAILED tests/models/flava/test_modeling_flava.py::FlavaForPreTrainingTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param flava.text_model.pooler.dense.weight
FAILED tests/models/fnet/test_modeling_fnet.py::FNetModelTest::test_training_gradient_checkpointing_autocast - RuntimeError: cuFFT only supports dimensions whose sizes are powers of two when computing in half precision, but got a signal size of[7, 32]
FAILED tests/models/gpt2/test_modeling_gpt2.py::GPT2ModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param multiple_choice_head.summary.weight
FAILED tests/models/graphormer/test_modeling_graphormer.py::GraphormerModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param encoder.graph_encoder.graph_attn_bias.edge_encoder.weight
FAILED tests/models/imagegpt/test_modeling_imagegpt.py::ImageGPTModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param transformer.h.0.ln_1.weight
FAILED tests/models/informer/test_modeling_informer.py::InformerModelTest::test_training_gradient_checkpointing_autocast - RuntimeError: Index put requires the source and destination dtypes match, got Float for the destination and Half for the source.
FAILED tests/models/layoutlm/test_modeling_layoutlm.py::LayoutLMModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param layoutlm.pooler.dense.weight
FAILED tests/models/lilt/test_modeling_lilt.py::LiltModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param lilt.encoder.layer.1.attention.self.layout_value.weight
FAILED tests/models/luke/test_modeling_luke.py::LukeModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param luke.pooler.dense.weight
FAILED tests/models/marian/test_modeling_marian.py::MarianModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param model.encoder.embed_positions.weight
FAILED tests/models/owlvit/test_modeling_owlvit.py::OwlViTVisionModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vision_model.post_layernorm.weight
FAILED tests/models/owlvit/test_modeling_owlvit.py::OwlViTForObjectDetectionTest::test_training_gradient_checkpointing_autocast - RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [12, 4, 64]], which is output 0 of AsStridedBackward0, is at version 1; expected version 0 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).
FAILED tests/models/pegasus/test_modeling_pegasus.py::PegasusModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param model.encoder.embed_positions.weight
FAILED tests/models/pix2struct/test_modeling_pix2struct.py::Pix2StructTextModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param layer.0.encoder_decoder_attention.attention.query.weight
FAILED tests/models/regnet/test_modeling_regnet.py::RegNetModelTest::test_training_gradient_checkpointing_autocast - RuntimeError: GET was unable to find an engine to execute this computation
FAILED tests/models/roformer/test_modeling_roformer.py::RoFormerModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param roformer.encoder.embed_positions.weight
FAILED tests/models/sam/test_modeling_sam.py::SamModelTest::test_training_gradient_checkpointing_autocast - AttributeError: 'SamModelTester' object has no attribute 'is_training'
FAILED tests/models/speech_to_text/test_modeling_speech_to_text.py::Speech2TextModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param model.encoder.embed_positions.weights
FAILED tests/models/switch_transformers/test_modeling_switch_transformers.py::SwitchTransformersModelTest::test_training_gradient_checkpointing_autocast - RuntimeError: Index put requires the source and destination dtypes match, got Float for the destination and Half for the source.
FAILED tests/models/time_series_transformer/test_modeling_time_series_transformer.py::TimeSeriesTransformerModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param model.encoder.embed_positions.weight
FAILED tests/models/van/test_modeling_van.py::VanModelTest::test_training_gradient_checkpointing_autocast - RuntimeError: GET was unable to find an engine to execute this computation
FAILED tests/models/vilt/test_modeling_vilt.py::ViltModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vilt.pooler.dense.weight
FAILED tests/models/visual_bert/test_modeling_visual_bert.py::VisualBertModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param visual_bert.pooler.dense.weight
FAILED tests/models/vit_mae/test_modeling_vit_mae.py::ViTMAEModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vit.embeddings.position_embeddings
FAILED tests/models/x_clip/test_modeling_x_clip.py::XCLIPVisionModelTest::test_training_gradient_checkpointing_autocast - AssertionError: False is not true : None gradient in param vision_model.post_layernorm.weight
```
It fails for most of vision models, the fix is intended for a quite unique and niche use case (autocast float16 + gradient checkpointing ) and it works for the targeted most usecases (large LLMs). I am not sure about the approach we should follow here (fix everything in this PR or no?), so would love to hear your thoughts here

Reply:
Did the test pass before?